crates/ra_lsp_server/tests/heavy_tests/main.rs

```diff
@@ -258,3 +259,65 @@ fn main() {}
         json!([]),
     );
 }
+
+#[test]
+fn test_missing_module_code_action_in_json_project() {
+    let tmp_dir = TempDir::new().unwrap();
+    let code = format!(
+        r#"
+//- rust-project.json
+{{ 
+    "roots": [ "{PATH}" ], 
+    "crates": [ {{ "root_module": "{PATH}/src/lib.rs", "deps": [], "edition": "2015" }} ] 
```

Comment:
This does not exercise logic for handling dependencies. Let's have a test with two crates which exercises goto definition across crates?

Reply:
I added a test that has a dependency. It isn't goto definition, but instead completion. I imagine that's probably fine?

Reply:
Yeah, completion is actually better, because goto def might accidentally work via indexes even without project structue. Let's then delete *this* tests: heavy tests are slow, so it's good idea to minimize the amount of them to keep test-suite fast. 

If we feel the need to test the logic of lowering `project.json` to a crate graph, we'd better write a unit-test style test for it, which doesn't to IO.